detectable/src/main/java/com/synopsys/integration/detectable/detectables/projectinspector/ProjectInspectorParser.java

```diff
@@ -85,29 +91,56 @@ public List<CodeLocation> processModules(JsonReader reader) throws IOException {
     public CodeLocation codeLocationFromModule(ProjectInspectorModule module) {
         Map<String, Dependency> lookup = new HashMap<>();
 
-        //build the map of all external ids
-        module.components.forEach((dependencyId, component) -> {
-            lookup.computeIfAbsent(dependencyId, missingId -> convertProjectInspectorDependency(component));
-        });
+        if(module.components != null) {
+            //build the map of all external ids
+            module.components.forEach((dependencyId, component) -> {
+                lookup.computeIfAbsent(dependencyId, missingId -> convertProjectInspectorDependency(component));
+            });
+
+            //and add them to the graph
+            DependencyGraph mutableDependencyGraph = new BasicDependencyGraph();
+            module.components.forEach((moduleDependencyId, moduleDependency) -> {
+                Dependency dependency = lookup.get(moduleDependencyId);
+                if (moduleDependency.inclusionType.equals("DIRECT")) {
+                    mutableDependencyGraph.addDirectDependency(dependency);
+                } else if (moduleDependency.inclusionType.equals("TRANSITIVE")) {
+                    moduleDependency.includedBy.forEach(includedBy -> {
+                        if (lookup.containsKey(includedBy.id)) {
+                            mutableDependencyGraph.addChildWithParent(dependency, lookup.get(includedBy.id));
+                        } else { //Theoretically should not happen according to PI devs. -jp
+                            throw new RuntimeException("An error occurred reading the project inspector output." +
+                                    " An unknown parent dependency was encountered '" + includedBy.id + "' while including dependency '" + moduleDependency.name + "'.");
+                        }
+                    });
+                }
+            });
+            return new CodeLocation(mutableDependencyGraph, new File(module.moduleFile));
+        } else {
+            logger.info("From Detect 9.5.0 and onwards, we will support the latest version of Project Inspector i.e. 2024.2.0. If you are using the detect.project.inspector.path property, please use the latest version mentioned.");
```

Comment:
This is a logger information or a warning if there is a mismatch of PI and Detect version when PI is passed through `detect.project.inspector.path` property.